protowhat/checks/check_logic.py

```diff
@@ -16,47 +16,42 @@ def multi(state, *tests):
     This function could be thought as an AND statement, since all tests it runs must pass
 
     Args:
-        state: State instance describing student and solution code. Can be omitted if used with Ex().
-        args: one or more sub-SCTs to run.
+        state: State instance describing student and solution code,  can be omitted if used with Ex()
+        tests: one or more sub-SCTs to run.
 
     :Example:
-        The SCT below runs run two has_code cases.. ::
+        The SCT below checks two has_code cases.. ::
 
             Ex().multi(has_code('SELECT'), has_code('WHERE'))
 
-        The SCT below checks that a SELECT statement has both a WHERE and LIMIT clause.. ::
+        The SCT below uses ``multi`` to 'branch out' to check that
+        the SELECT statement has both a WHERE and LIMIT clause.. ::
 
             Ex().check_node('SelectStmt', 0).multi(
                 check_edge('where_clause'),
                 check_edge('limit_clause')
             )
     """
-
-    for arg in tests:
-        # when input is a single test, make iterable
-        if callable(arg):
-            arg = [arg]
-
-        for test in arg:
-            # assume test is function needing a state argument
-            # partial state so reporter can test
-            test(state)
+    for test in iter_tests(tests):
+        # assume test is function needing a state argument
+        # partial state so reporter can test
+        state.do_test(partial(test, state))
 
     # return original state, so can be chained
     return state
 
 
-def check_not(state, *tests, incorrect_msg):
+def check_not(state, *tests, msg):
```

Comment:
Does this break anything?

Reply:
I found only one exercise uses it using our apidump cli.

As we want to make the interfaces more consistent over time, a decorator to redirect arguments could be useful.